opm/autodiff/NewtonIterationBlackoilInterleaved.hpp

```diff
@@ -122,9 +122,9 @@ namespace Opm
             EWOMS_REGISTER_PARAM(TypeTag, int, LinearSolverRestart, "The number of iterations after which GMRES is restarted");
             EWOMS_REGISTER_PARAM(TypeTag, int, FlowLinearSolverVerbosity, "The verbosity level of the linear solver (0: off, 2: all)");
             EWOMS_REGISTER_PARAM(TypeTag, int, IluFillinLevel, "The fill-in level of the linear solver's ILU preconditioner");
-            EWOMS_REGISTER_PARAM(TypeTag, std::string, MIluVariant, "Specify which variant of the modified-ILU preconditioner ought to be used");
+            EWOMS_REGISTER_PARAM(TypeTag, std::string, MiluVariant, "Specify which variant of the modified-ILU preconditioner ought to be used. Possible variants are: ILU (default, plain ILU), MILU_1 (lump diagonal with dropped row entries), MILU_2 (lump diagonal with the sum of the absolute values of the dropped row  entries), MILU_3 (if diagonal is positive add sum of dropped row entrires. Otherwise substract them), MILU_4 (if diagonal is positive add sum of dropped row entrires. Otherwise do nothing");
```

Comment:
this is going to be a looong line in the usage message.

Reply:
Then please tell me how to break it or better: point to the documentation of the command line parameter system.

Reply:
regarding the documentation of the parameter system: the whole API is these four macros:

https://github.com/OPM/ewoms/blob/master/ewoms/common/parametersystem.hh#L59-L119

if you want to write some prose on how they are used within eWoms and/or flow you're most welcome to do so.

Reply:
> https://github.com/OPM/ewoms/blob/master/ewoms/common/parametersystem.hh#L59-L119

I just took another look. But I still do not see how that explains my problems with the command line parameters or how to properly use them. I did not even use those macros in my code.

Reply:
> "Read the source, dude" is not documentation.

is "run Doxygen, dude" better? ;)

> Concerning the help page: If I do not get a list of possible values for a parameter, then the help is rather useless. BTW there are more parameters lacking this.

true. I've been thinking for a while that having proper support for "enumeration parameters" would be nice. It just never was a priority for me yet.

also, something like git's approach to help would be -- err -- helpful: print a as-brief-as-possible usage message for all parameters on `--help` and output all the gory on `--help $paramname`. adding this functionality and the necessary docstrings would be a medium-sized project, though.

> I just took another look. But I still do not see how that explains my problems with the command line parameters or how to properly use it.

I suppose that your issues are more related to how this machinery is used than the machinery itself. I don't have a good answer for that, maybe the best thing to do is to step through the parameter registration with a debugger.